### DISCUSSION TITLE
Training Issue

### DISCUSSION THREAD

**Discussion by jramborger78** [talmolab/sleap#1643] [Help!]
Hey guys,

I keep having this issue with trying to run a training. As my other discussion posts mention, I have a successful model that has been doing trackings. Recently, it was decided to now add another node to track the cue light precisely. Now it keeps having this issue. I have uninstalled and reinstalled, updated TF, cuda, cudnn, and still the issue. I tested it with the prior label project and it happened as well, so don't think it is the addition. Recently I updated my nvidia gpu driver, and that is my best guess, but i then reinstalled and reupdated everything and still occurring. Any ideas?

This happens at the start of the centroid training:
```
This message will be only logged once.
WARNING:tensorflow:Callback method `on_train_batch_end` is slow compared to the batch time (batch time: 0.0220s vs `on_train_batch_end` time: 0.0300s). Check your callbacks.
203/203 - 22s - loss: 4.2156e-04 - val_loss: 4.1083e-04
2023-12-18 16:19:19.018408: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.58GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:19.309097: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.58GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:19.757521: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.58GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:20.294141: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.56GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:20.294969: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.56GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:20.328988: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.56GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:20.329949: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.56GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:20.627494: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.56GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:21.437926: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.57GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:21.519130: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 1.57GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:19:21.532217: I tensorflow/stream_executor/cuda/cuda_blas.cc:1760] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.
Epoch 2/200
```

And this happens before the centered can before it quits:
```
Epoch 1/200
2023-12-18 16:46:48.634632: I tensorflow/stream_executor/cuda/cuda_dnn.cc:369] Loaded cuDNN version 8201
2023-12-18 16:46:50.317707: W tensorflow/stream_executor/gpu/redzone_allocator.cc:314] Internal: ptxas exited with non-zero error code -1, output:
Relying on driver to perform ptx compilation.
Modify $PATH to customize ptxas location.
This message will be only logged once.
2023-12-18 16:46:52.862467: W tensorflow/core/common_runtime/bfc_allocator.cc:338] Garbage collection: deallocate free memory regions (i.e., allocations) so that we can re-allocate a larger region to avoid OOM due to memory fragmentation. If you see this message frequently, you are running near the threshold of the available device memory and re-allocation may incur great performance overhead. You may try smaller batch sizes to observe the performance impact. Set TF_ENABLE_GPU_GARBAGE_COLLECTION=false if you'd like to disable this feature.
2023-12-18 16:46:53.258298: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 3.89GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:46:53.259461: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 3.89GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:46:53.357849: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 3.65GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:46:53.358156: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 3.65GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:46:53.493547: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 3.57GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:46:53.495195: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 3.57GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:46:53.616306: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 3.59GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:46:53.616830: W tensorflow/core/common_runtime/bfc_allocator.cc:272] Allocator (GPU_0_bfc) ran out of memory trying to allocate 3.59GiB with freed_by_count=0. The caller indicates that this is not a failure, but may mean that there could be performance gains if more memory were available.
2023-12-18 16:46:53.680946: W tensorflow/core/framework/op_kernel.cc:1680] Invalid argument: required broadcastable shapes
2023-12-18 16:46:53.681248: W tensorflow/core/framework/op_kernel.cc:1680] Invalid argument: required broadcastable shapes
Traceback (most recent call last):
  File "C:\mambaforge\envs\sleap\Scripts\sleap-train-script.py", line 33, in <module>
    sys.exit(load_entry_point('sleap==1.3.3', 'console_scripts', 'sleap-train')())
  File "C:\mambaforge\envs\sleap\lib\site-packages\sleap\nn\training.py", line 2014, in main
    trainer.train()
  File "C:\mambaforge\envs\sleap\lib\site-packages\sleap\nn\training.py", line 941, in train
    verbose=2,
  File "C:\Users\jramb\AppData\Roaming\Python\Python37\site-packages\keras\engine\training.py", line 1184, in fit
    tmp_logs = self.train_function(iterator)
  File "C:\Users\jramb\AppData\Roaming\Python\Python37\site-packages\tensorflow\python\eager\def_function.py", line 885, in __call__
    result = self._call(*args, **kwds)
  File "C:\Users\jramb\AppData\Roaming\Python\Python37\site-packages\tensorflow\python\eager\def_function.py", line 950, in _call
    return self._stateless_fn(*args, **kwds)
  File "C:\Users\jramb\AppData\Roaming\Python\Python37\site-packages\tensorflow\python\eager\function.py", line 3040, in __call__
    filtered_flat_args, captured_inputs=graph_function.captured_inputs)  # pylint: disable=protected-access
  File "C:\Users\jramb\AppData\Roaming\Python\Python37\site-packages\tensorflow\python\eager\function.py", line 1964, in _call_flat
    ctx, args, cancellation_manager=cancellation_manager))
  File "C:\Users\jramb\AppData\Roaming\Python\Python37\site-packages\tensorflow\python\eager\function.py", line 596, in call
    ctx=ctx)
  File "C:\Users\jramb\AppData\Roaming\Python\Python37\site-packages\tensorflow\python\eager\execute.py", line 60, in quick_execute
    inputs, attrs, num_outputs)
tensorflow.python.framework.errors_impl.InvalidArgumentError:  required broadcastable shapes
         [[node gradient_tape/loss_fn/mean_squared_error/sub (defined at mambaforge\envs\sleap\lib\site-packages\sleap\nn\training.py:941) ]] [Op:__inference_train_function_38927]

Function call stack:
train_function
```

**Comment by Marielamend** [comment]
Hello, @jramborger78!

To best assist you,

1. Can you send us your training config json?
2. Are you training from scratch or are you resuming training? If you are not training from scratch, the issue could be that some of the configurations are being carried over from the previous project.

Have a great day,
Mariela

**Comment by jramborger78** [comment]
That is exactly one i just thought about this morning and began from
scratch and so far it is running smoothly, just slower than before. I'll
keep you posted in the mean time! Here is the config, and just briefing it
it makes sense.

{
    "data": {
        "labels": {
            "training_labels": "C:/Users/jramb/Documents/Sleap/no_tail.slp",
            "validation_labels": null,
            "validation_fraction": 0.1,
            "test_labels": null,
            "split_by_inds": false,
            "training_inds": [
                497,
                736,
                741,
                243,
                787,
                978,
                10,
                408,
                162,
                757,
                561,
                810,
                915,
                464,
                586,
                654,
                395,
                803,
                555,
                550,
                681,
                599,
                394,
                104,
                778,
                64,
                721,
                640,
                628,
                108,
                532,
                69,
                752,
                743,
                422,
                543,
                920,
                401,
                945,
                67,
                895,
                522,
                847,
                301,
                679,
                593,
                388,
                289,
                51,
                508,
                655,
                667,
                385,
                788,
                137,
                342,
                443,
                851,
                785,
                462,
                198,
                587,
                563,
                621,
                624,
                637,
                992,
                361,
                557,
                43,
                845,
                664,
                817,
                435,
                52,
                125,
                208,
                825,
                224,
                109,
                784,
                160,
                354,
                468,
                952,
                729,
                303,
                789,
                669,
                450,
                196,
                250,
                493,
                520,
                818,
                921,
                60,
                657,
                811,
                121,
                326,
                219,
                94,
                460,
                467,
                829,
                461,
                324,
                606,
                359,
                38,
                390,
                855,
                316,
                666,
                511,
                320,
                330,
                802,
                777,
                106,
                414,
                228,
                381,
                875,
                403,
                881,
                335,
                202,
                372,
                638,
                391,
                55,
                516,
                523,
                694,
                975,
                268,
                118,
                131,
                482,
                956,
                251,
                409,
                627,
                916,
                993,
                111,
                249,
                739,
                605,
                193,
                764,
                46,
                130,
                935,
                942,
                77,
                28,
                927,
                642,
                656,
                941,
                270,
                20,
                238,
                484,
                266,
                488,
                631,
                132,
                363,
                312,
                500,
                548,
                630,
                378,
                918,
                890,
                257,
                573,
                917,
                16,
                999,
                89,
                888,
                754,
                427,
                575,
                37,
                454,
                32,
                797,
                841,
                96,
                974,
                768,
                293,
                158,
                12,
                174,
                226,
                430,
                673,
                690,
                965,
                27,
                321,
                731,
                383,
                366,
                538,
                417,
                431,
                314,
                25,
                66,
                7,
                14,
                931,
                948,
                386,
                871,
                929,
                683,
                903,
                338,
                769,
                54,
                850,
                133,
                505,
                913,
                127,
                874,
                529,
                724,
                933,
                686,
                723,
                189,
                339,
                1001,
                180,
                677,
                900,
                159,
                821,
                451,
                982,
                562,
                182,
                480,
                632,
                186,
                400,
                418,
                134,
                200,
                216,
                746,
                899,
                362,
                792,
                248,
                726,
                553,
                205,
                50,
                113,
                370,
                291,
                300,
                534,
                990,
                296,
                30,
                83,
                105,
                317,
                684,
                822,
                433,
                760,
                961,
                889,
                753,
                232,
                345,
                164,
                831,
                758,
                192,
                494,
                674,
                834,
                489,
                558,
                622,
                13,
                360,
                706,
                275,
                801,
                379,
                755,
                928,
                989,
                564,
                473,
                325,
                139,
                44,
                843,
                466,
                693,
                824,
                35,
                365,
                891,
                141,
                932,
                197,
                717,
                347,
                369,
                823,
                909,
                70,
                737,
                311,
                272,
                432,
                242,
                319,
                566,
                290,
                269,
                820,
                504,
                149,
                830,
                761,
                510,
                525,
                179,
                672,
                977,
                710,
                225,
                957,
                539,
                128,
                506,
                92,
                2,
                410,
                145,
                40,
                771,
                781,
                647,
                259,
                880,
                840,
                499,
                434,
                110,
                814,
                191,
                873,
                144,
                806,
                997,
                725,
                645,
                47,
                696,
                866,
                893,
                984,
                971,
                623,
                445,
                959,
                438,
                0,
                442,
                815,
                652,
                457,
                727,
                687,
                567,
                930,
                625,
                455,
                914,
                292,
                646,
                173,
                836,
                919,
                512,
                614,
                176,
                253,
                712,
                476,
                991,
                634,
                552,
                770,
                304,
                626,
                492,
                161,
                288,
                331,
                368,
                639,
                490,
                572,
                697,
                305,
                491,
                340,
                59,
                206,
                475,
                404,
                18,
                574,
                772,
                530,
                783,
                507,
                711,
                868,
                463,
                503,
                776,
                472,
                63,
                155,
                950,
                633,
                745,
                648,
                597,
                235,
                447,
                635,
                280,
                565,
                285,
                184,
                700,
                735,
                604,
                142,
                730,
                680,
                76,
                812,
                600,
                11,
                643,
                966,
                486,
                479,
                502,
                939,
                882,
                308,
                116,
                23,
                846,
                759,
                618,
                595,
                84,
                662,
                827,
                589,
                692,
                691,
                165,
                156,
                233,
                960,
                869,
                641,
                136,
                862,
                175,
                346,
                283,
                607,
                722,
                720,
                171,
                48,
                924,
                912,
                41,
                167,
                78,
                223,
                732,
                892,
                98,
                138,
                994,
                22,
                779,
                867,
                88,
                279,
                947,
                436,
                603,
                906,
                904,
                188,
                570,
                129,
                263,
                33,
                124,
                203,
                578,
                221,
                143,
                299,
                343,
                309,
                471,
                569,
                616,
                333,
                554,
                470,
                620,
                782,
                260,
                524,
                742,
                103,
                72,
                483,
                245,
                147,
                375,
                392,
                714,
                668,
                894,
                854,
                87,
                650,
                485,
                576,
                29,
                968,
                705,
                428,
                254,
                987,
                848,
                439,
                910,
                261,
                860,
                963,
                306,
                908,
                398,
                240,
                790,
                582,
                537,
                68,
                277,
                651,
                728,
                738,
                838,
                209,
                615,
                826,
                123,
                5,
                685,
                122,
                252,
                234,
                15,
                546,
                853,
                519,
                429,
                816,
                926,
                153,
                170,
                983,
                911,
                611,
                878,
                610,
                241,
                297,
                740,
                374,
                828,
                707,
                976,
                195,
                58,
                26,
                719,
                446,
                8,
                329,
                560,
                972,
                698,
                194,
                177,
                215,
                922,
                544,
                839,
                352,
                969,
                660,
                897,
                954,
                91,
                568,
                980,
                617,
                351,
                231,
                402,
                353,
                426,
                276,
                214,
                786,
                456,
                704,
                19,
                6,
                382,
                898,
                946,
                585,
                584,
                613,
                699,
                979,
                695,
                267,
                411,
                222,
                592,
                201,
                65,
                531,
                775,
                795,
                547,
                157,
                813,
                886,
                21,
                715,
                114,
                281,
                237,
                955,
                885,
                415,
                986,
                901,
                713,
                670,
                440,
                190,
                481,
                528,
                762,
                265,
                217,
                407,
                943,
                594,
                421,
                477,
                857,
                663,
                571,
                748,
                864,
                747,
                527,
                377,
                373,
                172,
                808,
                649,
                424,
                665,
                951,
                318,
                389,
                295,
                53,
                82,
                858,
                509,
                591,
                832,
                833,
                859,
                767,
                376,
                517,
                49,
                588,
                358,
                676,
                75,
                126,
                416,
                708,
                187,
                453,
                887,
                925,
                763,
                590,
                302,
                9,
                102,
                883,
                211,
                953,
                773,
                307,
                419,
                230,
                204,
                905,
                210,
                549,
                835,
                487,
                459,
                689,
                258,
                577,
                264,
                865,
                981,
                793,
                256,
                74,
                73,
                323,
                356,
                958,
                733,
                441,
                62,
                99,
                598,
                805,
                800,
                262,
                861,
                612,
                107,
                601,
                716,
                85,
                95,
                937,
                967,
                580,
                469,
                229,
                17,
                988,
                907,
                168,
                255,
                940,
                514,
                423,
                452,
                718,
                42,
                278,
                246,
                478,
                682,
                380,
                765,
                24,
                405,
                146,
                355,
                218,
                938,
                774,
                449,
                154,
                239,
                425,
                166,
                294,
                120,
                336,
                152,
                884,
                444,
                458,
                474,
                247,
                178,
                332,
                837,
                327,
                115,
                799,
                364,
                842,
                671,
                61,
                661,
                513,
                71,
                794,
                809,
                286,
                4,
                596,
                996,
                31,
                542,
                609,
                872,
                350,
                541,
                581,
                57,
                334,
                341,
                39,
                100,
                583,
                856,
                1,
                515,
                653,
                437,
                796,
                804,
                140,
                756,
                501,
                212,
                45,
                985,
                619,
                244,
                465,
                540,
                902,
                220,
                119,
                535,
                397,
                536,
                284,
                556,
                998,
                413,
                207,
                879,
                751,
                315,
                644,
                852,
                629,
                271,
                36,
                602,
                749
            ],
            "validation_inds": [
                227,
                545,
                709,
                213,
                608,
                81,
                636,
                521,
                420,
                313,
                659,
                780,
                310,
                964,
                399,
                349,
                185,
                298,
                798,
                181,
                533,
                973,
                896,
                112,
                80,
                357,
                183,
                101,
                791,
                406,
                701,
                287,
                944,
                559,
                322,
                877,
                658,
                367,
                93,
                934,
                863,
                498,
                117,
                551,
                236,
                348,
                151,
                526,
                766,
                282,
                448,
                90,
                703,
                579,
                97,
                79,
                688,
                328,
                150,
                744,
                163,
                870,
                135,
                807,
                750,
                495,
                34,
                396,
                169,
                3,
                337,
                86,
                387,
                844,
                970,
                995,
                56,
                734,
                199,
                412,
                678,
                819,
                1000,
                936,
                148,
                384,
                675,
                923,
                273,
                496,
                393,
                518,
                962,
                949,
                849,
                344,
                876,
                274,
                702,
                371
            ],
            "test_inds": null,
            "search_path_hints": [
                "",
                "",
                "",
                "",
                "",
                "",
                "",
                "",
                ""
            ],
            "skeletons": [
                {
                    "directed": true,
                    "graph": {
                        "name": "Skeleton-1",
                        "num_edges_inserted": 18
                    },
                    "links": [
                        {
                            "edge_insert_idx": 0,
                            "key": 0,
                            "source": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Neck",
                                        1.0
                                    ]
                                }
                            },
                            "target": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Nose",
                                        1.0
                                    ]
                                }
                            },
                            "type": {
                                "py/reduce": [
                                    {
                                        "py/type": "sleap.skeleton.EdgeType"
                                    },
                                    {
                                        "py/tuple": [
                                            1
                                        ]
                                    }
                                ]
                            }
                        },
                        {
                            "edge_insert_idx": 1,
                            "key": 0,
                            "source": {
                                "py/id": 1
                            },
                            "target": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Ear_L",
                                        1.0
                                    ]
                                }
                            },
                            "type": {
                                "py/id": 3
                            }
                        },
                        {
                            "edge_insert_idx": 2,
                            "key": 0,
                            "source": {
                                "py/id": 1
                            },
                            "target": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Ear_R",
                                        1.0
                                    ]
                                }
                            },
                            "type": {
                                "py/id": 3
                            }
                        },
                        {
                            "edge_insert_idx": 3,
                            "key": 0,
                            "source": {
                                "py/id": 1
                            },
                            "target": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Front_L",
                                        1.0
                                    ]
                                }
                            },
                            "type": {
                                "py/id": 3
                            }
                        },
                        {
                            "edge_insert_idx": 4,
                            "key": 0,
                            "source": {
                                "py/id": 1
                            },
                            "target": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Front_R",
                                        1.0
                                    ]
                                }
                            },
                            "type": {
                                "py/id": 3
                            }
                        },
                        {
                            "edge_insert_idx": 14,
                            "key": 0,
                            "source": {
                                "py/id": 1
                            },
                            "target": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Catheter",
                                        1.0
                                    ]
                                }
                            },
                            "type": {
                                "py/id": 3
                            }
                        },
                        {
                            "edge_insert_idx": 16,
                            "key": 0,
                            "source": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Base",
                                        1.0
                                    ]
                                }
                            },
                            "target": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Back_L",
                                        1.0
                                    ]
                                }
                            },
                            "type": {
                                "py/id": 3
                            }
                        },
                        {
                            "edge_insert_idx": 17,
                            "key": 0,
                            "source": {
                                "py/id": 9
                            },
                            "target": {
                                "py/object": "sleap.skeleton.Node",
                                "py/state": {
                                    "py/tuple": [
                                        "Back_R",
                                        1.0
                                    ]
                                }
                            },
                            "type": {
                                "py/id": 3
                            }
                        },
                        {
                            "edge_insert_idx": 15,
                            "key": 0,
                            "source": {
                                "py/id": 8
                            },
                            "target": {
                                "py/id": 9
                            },
                            "type": {
                                "py/id": 3
                            }
                        }
                    ],
                    "multigraph": true,
                    "nodes": [
                        {
                            "id": {
                                "py/id": 2
                            }
                        },
                        {
                            "id": {
                                "py/id": 4
                            }
                        },
                        {
                            "id": {
                                "py/id": 5
                            }
                        },
                        {
                            "id": {
                                "py/id": 1
                            }
                        },
                        {
                            "id": {
                                "py/id": 6
                            }
                        },
                        {
                            "id": {
                                "py/id": 7
                            }
                        },
                        {
                            "id": {
                                "py/id": 9
                            }
                        },
                        {
                            "id": {
                                "py/id": 10
                            }
                        },
                        {
                            "id": {
                                "py/id": 11
                            }
                        },
                        {
                            "id": {
                                "py/id": 8
                            }
                        }
                    ]
                }
            ]
        },
        "preprocessing": {
            "ensure_rgb": false,
            "ensure_grayscale": false,
            "imagenet_mode": null,
            "input_scaling": 1.0,
            "pad_to_stride": 1,
            "resize_and_pad_to_target": true,
            "target_height": 600,
            "target_width": 800
        },
        "instance_cropping": {
            "center_on_part": "Catheter",
            "crop_size": 320,
            "crop_size_detection_padding": 16
        }
    },
    "model": {
        "backbone": {
            "leap": null,
            "unet": {
                "stem_stride": null,
                "max_stride": 64,
                "output_stride": 4,
                "filters": 24,
                "filters_rate": 2.0,
                "middle_block": true,
                "up_interpolate": true,
                "stacks": 1
            },
            "hourglass": null,
            "resnet": null,
            "pretrained_encoder": null
        },
        "heads": {
            "single_instance": null,
            "centroid": null,
            "centered_instance": {
                "anchor_part": "Catheter",
                "part_names": [
                    "Nose",
                    "Ear_L",
                    "Ear_R",
                    "Neck",
                    "Front_L",
                    "Front_R",
                    "Base",
                    "Back_L",
                    "Back_R",
                    "Catheter"
                ],
                "sigma": 2.0,
                "output_stride": 4,
                "loss_weight": 1.0,
                "offset_refinement": false
            },
            "multi_instance": null,
            "multi_class_bottomup": null,
            "multi_class_topdown": null
        },
        "base_checkpoint": null
    },
    "optimization": {
        "preload_data": true,
        "augmentation_config": {
            "rotate": true,
            "rotation_min_angle": -180.0,
            "rotation_max_angle": 180.0,
            "translate": false,
            "translate_min": -5,
            "translate_max": 5,
            "scale": false,
            "scale_min": 0.9,
            "scale_max": 1.1,
            "uniform_noise": false,
            "uniform_noise_min_val": 0.0,
            "uniform_noise_max_val": 10.0,
            "gaussian_noise": false,
            "gaussian_noise_mean": 5.0,
            "gaussian_noise_stddev": 1.0,
            "contrast": false,
            "contrast_min_gamma": 0.5,
            "contrast_max_gamma": 2.0,
            "brightness": false,
            "brightness_min_val": 0.0,
            "brightness_max_val": 10.0,
            "random_crop": false,
            "random_crop_height": 256,
            "random_crop_width": 256,
            "random_flip": false,
            "flip_horizontal": false
        },
        "online_shuffling": true,
        "shuffle_buffer_size": 128,
        "prefetch": true,
        "batch_size": 4,
        "batches_per_epoch": 214,
        "min_batches_per_epoch": 200,
        "val_batches_per_epoch": 11,
        "min_val_batches_per_epoch": 10,
        "epochs": 200,
        "optimizer": "adam",
        "initial_learning_rate": 0.0001,
        "learning_rate_schedule": {
            "reduce_on_plateau": true,
            "reduction_factor": 0.5,
            "plateau_min_delta": 1e-06,
            "plateau_patience": 5,
            "plateau_cooldown": 3,
            "min_learning_rate": 1e-08
        },
        "hard_keypoint_mining": {
            "online_mining": false,
            "hard_to_easy_ratio": 2.0,
            "min_hard_keypoints": 2,
            "max_hard_keypoints": null,
            "loss_scale": 5.0
        },
        "early_stopping": {
            "stop_training_on_plateau": true,
            "plateau_min_delta": 1e-08,
            "plateau_patience": 10
        }
    },
    "outputs": {
        "save_outputs": true,
        "run_name": "231203_181655.centered_instance.n=1002",
        "run_name_prefix": "",
        "run_name_suffix": "",
        "runs_folder": "C:/Users/jramb/Documents/Sleap\\models",
        "tags": [
            ""
        ],
        "save_visualizations": true,
        "delete_viz_images": true,
        "zip_outputs": false,
        "log_to_csv": true,
        "checkpointing": {
            "initial_model": false,
            "best_model": true,
            "every_epoch": false,
            "latest_model": false,
            "final_model": false
        },
        "tensorboard": {
            "write_logs": false,
            "loss_frequency": "epoch",
            "architecture_graph": false,
            "profile_graph": false,
            "visualizations": true
        },
        "zmq": {
            "subscribe_to_controller": true,
            "controller_address": "tcp://127.0.0.1:9000",
            "controller_polling_timeout": 10,
            "publish_updates": true,
            "publish_address": "tcp://127.0.0.1:9001"
        }
    },
    "name": "",
    "description": "",
    "sleap_version": "1.3.3",
    "filename":
"C:/Users/jramb/Documents/Sleap\\models\\231203_181655.centered_instance.n=1002\\training_config.json"
}

On Tue, Dec 19, 2023 at 11:06 AM Marielamend ***@***.***>
wrote:

> Hello, @jramborger78 <https://github.com/jramborger78>!
>
> To best assist you,
>
>    1. Can you send us your training config json?
>    2. Are you training from scratch or are you resuming training? If you
>    are not training from scratch, the issue could be that some of the
>    configurations are being carried over from the previous project.
>
> Have a great day,
> Mariela
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/talmolab/sleap/discussions/1643#discussioncomment-7900612>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BABV7SOPDEPE7FO5KAKNSELYKHJR3AVCNFSM6AAAAABA2JZFYCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TSMBQGYYTE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Sincerely,

Jarryd Ramborger
951-541-3922
